### PR TITLE
added cli

### PR DESCRIPTION
**Pull request by dabit3**
Added basic cli using existing app files and file structure.

To create new app:
1. Install create reason react app globally
`npm i -g create-reason-react-app`

2. Create new project by passing in name as first argument:
`create-reason-react-app MyReasonApp`

Thanks @knowbody , very nice project.

Thanks to @binoy14 for showing me `cross-spawn` yesterday.

### CODE REVIEW

**Comment by dabit3** [comment]
@knowbody Let me know if you run into any problems testing.

To test:

1. clone project

2. cd into bin directory and run `npm link`

3. Run `create-reason-react-app MyProject` command

File: package.json [hunk 1]
```diff
@@ -1,55 +1,25 @@
 {
   "name": "create-reason-react-app",
-  "private": true,
-  "version": "0.0.0",
-  "description": "",
+  "version": "0.1.0",
+  "description": "reason react project generator",
```
Comment (knowbody) [inline package.json]: Can we name this Create Reason React App?
and the other `package.json` - Reason React App

File: bin/cli.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+program
+.version(package.version)
+  .usage('<keywords>')
+  .option('-n, --new, [name]', 'Create the application')
+  .parse(process.argv);
+
+if (program.args.length === 1) {
+  spawn(build(program.args[0]), { shell: true, stdio: 'inherit' });
+} else if (program.args.length < 1) {
+  console.log(chalk.red('Please supply a name for your new React Reason app.'));
```
Comment (knowbody) [inline bin/cli.js]: typo: Reason React

File: bin/build.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      }
+      resolve();
+    })
+  })
+}
+
+const build = (appName) => {
+  cp('-r', __dirname + '/../src/', appName);
+  console.log('----------------------------------------------------------');
+  figlet.text('react reason', function(err, data) {
```
Comment (knowbody) [inline bin/build.js]: typo: reason react

**Comment by knowbody** [comment]
@dabit3 Also could we move the `/guides` out of `/src` and keep it in the root of the repo, please.

Besides these few typos, this is awesome! Thank you for the PR 🎉

**Comment by dabit3** [comment]
@knowbody ok, let me know if those updates look right.

**Comment by knowbody** [comment]
Great! thanks a lot @dabit3

**Pull request merged**